src/docfx/build/redirection/RedirectionProvider.cs

```diff
@@ -70,6 +74,12 @@ public string GetRedirectUrl(ErrorBuilder errors, FilePath file)
             while (_history.Value.redirectionHistory.TryGetValue(redirectionFile, out var item))
             {
                 var (renamedFrom, source) = item;
+                var dir = Path.GetDirectoryName(source?.File.Path);
+                if (source != null && !string.IsNullOrEmpty(dir))
+                {
+                    renamedFrom = renamedFrom.WithPath(new PathString(Path.Combine(dir, renamedFrom.Path)));
```

Comment:
is it possible to make the path right in `GetRenameAndRedirectionHistory`?

Reply:
Didn't do that because we need the original content to report warnings/errors later.

Reply:
@AngryBerryMS , keeping every file path in member variable relative to docset is a good practice, the error would be reported when building the redirection map.